package/MDAnalysis/lib/qcprot.pyx

```diff
@@ -186,15 +192,15 @@ def InnerProduct(np.ndarray[np.float64_t,ndim=1] A,
 
     if (weight is not None):
         for i in range(N):
-            x1 = weight[i] * coords1[0,i]
-            y1 = weight[i] * coords1[1,i]
-            z1 = weight[i] * coords1[2,i]
+            x1 = weight[i] * coords1[i,0]
```

Comment:
what is the memory layout used in MDAnalysis for arrays normally. For this code to be efficient it should be fortran style (row mayor). You can check the memory layout of an array with `some_array.flags`.